content/aspdotnet-controllers-and-routes/reading/controllers-forms/index.md

```diff
@@ -0,0 +1,126 @@
+---
+title: "Controllers with Forms"
+date: 2023-02-06T14:09:15-06:00
+draft: false
+weight: 3
+originalAuthor: Sally Steuterman # to be set by page creator
+originalAuthorGitHub: gildedgardenia # to be set by page creator
+reviewer: # to be set by the page reviewer
+reviewerGitHub: # to be set by the page reviewer
+lastEditor: # update any time edits are made after review
+lastEditorGitHub: # update any time edits are made after review
+lastMod: # UPDATE ANY TIME CHANGES ARE MADE
+---
+
+## Sending Form Data
+
+What if we want to send over some form data?
+To send data via a simple form in ASP.NET, we'll set things up like this:
+
+We have an action method that generates a form at index and responds to a `GET` request. 
+We will modify the `Index()` method we have already written like so and comment out any attributes we have been using:
+
+```csharp {linenos = table}
+   public IActionResult Index()
+   {
+      string html = "<form method='post' action='/hello/welcome'>" +
+         "<input type='text' name='name' />" +
+         "<input type='submit' value='Greet Me!' />" +
+         "</form>";
+
+      return Content(html, "text/html");
+   }
+```
+
+With conventional routing, we know that the `Index()` method currently responds to `localhost:5001/hello`.
+On form submission, we want to send the data down another path, `/hello/welcome`, which we specified in the `action` attribute in the `form` tag.
+This is the path to our `Welcome()` method.
+On line 4 above, we have given the text input the same name as the argument of the `Welcome()` method, which happens to be `name`.
+By giving the route to the `Welcome()` method in `action`, specifying `POST` as the `method` attribute, and matching the argument to the input, we have set up the `Welcome()` method to handle the form submission.
+
+With the `Index()` method rewritten and our `Welcome()` method handling form submission, we can run our app and see what happens!
+
+![Webpage with filled out form](pictures/filledoutform.png)
```

Comment:
There is a lot of white space with the photos on this page. Suggestion of cropping them.

Reply:
Re-sized as per my previous note! Thank you so much for the suggestion!